sdk/src/main/java/com/google/cloud/dataflow/sdk/util/Reshuffle.java

```diff
@@ -60,11 +54,14 @@ private Reshuffle() {
     // If the input has already had its windows merged, then the GBK that performed the merge
     // will have set originalStrategy.getWindowFn() to InvalidWindows, causing the GBK contained
     // here to fail. Instead, we install a valid WindowFn that leaves all windows unchanged.
-    Window.Bound<KV<K, V>> rewindow = Window
-        .<KV<K, V>>into(new PassThroughWindowFn<>(originalStrategy.getWindowFn()))
-        .triggering(new ReshuffleTrigger<>())
-        .discardingFiredPanes()
-        .withAllowedLateness(Duration.millis(BoundedWindow.TIMESTAMP_MAX_VALUE.getMillis()));
+    Window.Bound<KV<K, V>> rewindow =
+        Window.<KV<K, V>>into(
+                new IdentityWindowFn<>(
+                    originalStrategy.getWindowFn().windowCoder(),
+                    originalStrategy.getWindowFn().assignsToSingleWindow()))
+            .triggering(new ReshuffleTrigger<>())
+            .discardingFiredPanes()
+            .withAllowedLateness(Duration.millis(BoundedWindow.TIMESTAMP_MAX_VALUE.getMillis()));
```

Comment:
This is simply a backport of c9a1efae25455af1e47675938e296c716ec0fa0f. My only changes are package name and merge cleanup. I'm not intimately familiar with the details so I'll defer validation questions to @tgroh as the original author.

Note that this backport is not blocking other work but @kennknowles mentioned it will eventually be needed for PAssert.